File: code/graphs/dfs-iterative.cpp
void dfs(Graph& g, int s) {
    stack<int> st; st.push(s);
    vector<bool> seen(g.V, false);
    while (!st.empty()) {
        int u = st.top(); st.pop();
        if (seen[u]) continue;
        seen[u] = true;             // visit u
        for (int v : g.adj[u])
            if (!seen[v]) st.push(v);
    }
}

File: code/graphs/dfs-iterative.rs
fn dfs(g: &Graph, s: usize) {
    let mut st = vec![s];
    let mut seen = vec![false; g.adj.len()];
    while let Some(u) = st.pop() {
        if seen[u] { continue; }
        seen[u] = true;             // visit u
        for &v in &g.adj[u] {
            if !seen[v] { st.push(v); }
        }
    }
}
